services/order.py

```diff
@@ -0,0 +1,35 @@
+from typing import List, Dict, Optional
+from django.db import transaction
+from db.models import Order, Ticket, MovieSession
+from django.contrib.auth import get_user_model
+
+
+@transaction.atomic
+def create_order(
+    tickets: List[Dict[str, int]], username: str, date: Optional[str] = None
+) -> Order:
+    user = get_user_model().objects.get(username=username)
+    order = Order.objects.create(user=user)
+
+    ticket_objects: List[Ticket] = [
+        Ticket(
+            order=order,
+            row=data["row"],
+            seat=data["seat"],
+            movie_session=MovieSession.objects.get(id=data["movie_session"]),
+        )
+        for data in tickets
+    ]
+
+    if date:
+        order.created_at = date
```

Comment:
The `date` parameter is expected to be a string, but it should be a `datetime` object to match the model's field type. Ensure that the input is properly converted before saving.